internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -217,15 +217,15 @@ func WithManagedFieldsUpgrader(u ManagedFieldsUpgrader) FunctionComposerOption {
 
 // NewFunctionComposer returns a new Composer that supports composing resources using
 // both Patch and Transform (P&T) logic and a pipeline of Composition Functions.
-func NewFunctionComposer(kube client.Client, r FunctionRunner, o ...FunctionComposerOption) *FunctionComposer {
+func NewFunctionComposer(kube client.Client, nckube client.Client, r FunctionRunner, o ...FunctionComposerOption) *FunctionComposer {
```

Comment:
```suggestion
func NewFunctionComposer(cached, uncached client.Client, r FunctionRunner, o ...FunctionComposerOption) *FunctionComposer {
```

Nit: You can "stack" the type declaration in Go when you have several arguments of the same type in a row like this. I also prefer `cached` and `uncached` as names now that there's two clients for different reasons.

Reply:
Done